src/util-affinity.c

```diff
@@ -320,7 +320,39 @@ uint16_t UtilAffinityGetAffinedCPUNum(ThreadsAffinityType *taf)
     return ncpu;
 }
 
-#ifdef HAVE_DPDK
+/**
+ * \brief Verify that the CPU requirement is met
+ * \retval 0 if the requirement is met, 2 if the test is skipped, negative number otherwise
+ */
+int UnitTestsUtilAffinityVerifyCPURequirement(void)
```

Comment:
can we run this check before registering tests that would be skipped? That would keep the other logic simpler

Reply:
No, and I don't think that's a good idea. 

The check cannot be done because the thread affinity configuration of individual tests determines the number of required CPU cores. This can be redone to e.g. specify the number of required CPU cores by the test when it is being registered but that would specify the same information in two places. 
Maybe it would be possible if the input configuration is pulled out of the test so it is accessible when registering the test? but that just sounds tricky - something like `UtRegisterTestWithInput("DPDKRunmodeSetThreads01", DPDKRunmodeSetThreads01, input_text);`

Additionally, I don't think that's a good idea because if you don't register the test, the user/developer will never have an idea that such test exists. By saying it was skipped you at least present the information that more tests are available - if your aim was not remove the Skip test

Reply:
Based on what factor will the test be skipped in the PR? A system factor?

Reply:
yes, internally, some functions use calls to retrieve the number of CPU cores - it may happen that the number of configured threads is actually higher than the number of online CPU cores. I made the tests so they require <=3 cores but I thought it may sometime land on a docker container with e.g. only one/two cores.

Reply:
I guess, it would be possible to parametrize the internal functions to avoid "test skipping" by adding an extra parameter that would signify the number of available/online CPUs. That could make it more unit test friendly I guess.

Reply:
Let's continue in #10838